R/draws-index.R

```diff
@@ -173,6 +175,31 @@ set_variables <- function(x, variables, ...) {
   return(x)
 }
 
+#' @rdname set_variables
+#' @param old_variables (character) Variable names to repair. Should be variable
+#'   names with periods to convert to square brackets (e.g., `"theta.1"` ->
+#'   `"theta[1]"`, `"theta.1.1"` -> `"theta[1,1]"`).
+#' @examples
+#' # using repair_variable_names
+#' x <- matrix(rnorm(100), ncol = 2)
+#' colnames(x) <- c("theta.1", "theta.2")
+#' repair_variable_names(colnames(x))
+#' x <- set_variables(as_draws(x), repair_variable_names(colnames(x)))
+#' variables(x)
+#'
+#' @export
+repair_variable_names <- function(old_variables) {
+  if (!all(grepl("\\.", old_variables))) {
+    stop_no_call(
+      "All names in 'old_variables' must contain at least one '.' in the name."
+    )
```

Comment:
For simplicity I added an error if all the names don't have a `.` in it. It would be strange to pass in a vector like `c("theta.1", "theta[2]")` so I don't think we need to write out the logic to handle mixed cases like that. But let me know if you disagree and I can add that logic.

Reply:
This logic seems very unhelpful if I have some variables which are scalars. I can no longer just pass the entire list to this function 

I believe the set of transformations here is idempotent, so if someone passes `theta[1]`, nothing will be changed, so why not allow it